### PR TITLE
fix(styles): deal with new sass restriction

### LINKED ISSUE #13694: [Bug Report][2.5.0] New Sass Deprecation Recommended
### Environment
**Vuetify Version:** 2.5.0
**Vue Version:** 2.6.12
**Browsers:** Chrome 90.0.4430.212
**OS:** Windows 10

### Steps to reproduce
npm run build 

### Expected Behavior
Should not give warnings about sass

### Actual Behavior
gives a lot of recommendations warnings of near future deprecated syntax

### Reproduction Link
<a href="https://codesandbox.io/s/nuxtjs-vuetify-forked-nrfuw" target="_blank">https://codesandbox.io/s/nuxtjs-vuetify-forked-nrfuw</a>

### PR DESCRIPTION
**Pull request by mcdmaster**
cf. https://sass-lang.com/documentation/breaking-changes/slash-div
To deal with this new SASS anti-slash-div facility

Fixes #13694
Fixes https://github.com/tokyo-metropolitan-gov/covid19/issues/6346

## Motivation and Context

Other applications implementing SASS plus Vuetify are impacted on the change done by SASS

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [ ] The PR title is no longer than 64 characters.
- [ ] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [ ] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
I checked the build on both this and master with my update and the files are identical. With that being said, I'd still like @KaelWD to take a look at this.

Otherwise, looks good to me.

File: packages/vuetify/src/components/VCalendar/mixins/__tests__/calendar-with-events.spec.ts [hunk 1]
```diff
@@ -124,10 +124,10 @@ describe('calendar-with-events.ts', () => {
...
     // (12-hour with `h` || 12-hour with `hh` || 24-hour with `h` || 24-hour with `hh`),
     // we expect the value passed to be-
-    expect(wrapper.vm.formatTime(testData1, true)).toMatch(/^0?8:30( AM)?$/i) // 8:30 AM || 08:30 AM || 8:30 || 08:30
-    expect(wrapper.vm.formatTime(testData2, true)).toMatch(/^(0?5:45 PM|17:45)$/i) // 5:45 PM || 05:45 PM || 17:45
-    expect(wrapper.vm.formatTime(testData3, true)).toMatch(/^0?9:05( AM)?$/i) // 9:05 AM || 09:05 AM || 9:05 || 09:45
-    expect(wrapper.vm.formatTime(testData4, true)).toMatch(/^(0?3 PM|15)$/i) // 3 AM || 03 AM || 15
+    expect(wrapper.vm.formatTime(testData1, true)).toMatch(/^(午前)?0?8(時|:)30(分| AM)?$/i) // 8:30 AM || 08:30 AM || 8:30 || 08:30
+    expect(wrapper.vm.formatTime(testData2, true)).toMatch(/^(午後)?(0?5(時|:)45(分| PM)?|17:45)$/i) // 5:45 PM || 05:45 PM || 17:45
+    expect(wrapper.vm.formatTime(testData3, true)).toMatch(/^(午前)?0?9[時:]05(分| AM)?$/i) // 9:05 AM || 09:05 AM || 9:05 || 09:45
+    expect(wrapper.vm.formatTime(testData4, true)).toMatch(/^(午後)?(0?3[時:](PM)?)|(15時?)$/i) // 3 AM || 03 AM || 15
```
Comment (KaelWD) [inline packages/vuetify/src/components/VCalendar/mixins/__tests__/calendar-with-events.spec.ts:130]: Please remove unrelated changes

File: packages/vuetify/src/components/VGrid/_grid.sass [hunk 2]
```diff
@@ -73,9 +73,9 @@
     $n: $grid-columns
     @while $n > 0
       .flex.#{$size}#{$n}
-        flex-basis: $n / $grid-columns * 100%
+        flex-basis: calc(#{$n / $grid-columns} * 100%)
```
Comment (KaelWD) [inline packages/vuetify/src/components/VGrid/_grid.sass:76]: Don't use calc
```suggestion
        flex-basis: math.div($n, $grid-columns) * 100%
```

**Review by KaelWD — CHANGES_REQUESTED** [review]
Use `math.div` instead of muliplication

**Comment by KaelWD** [comment]
> we should probably opt to wrap our division in `#{}`

This is deprecated too.

**Review by KaelWD — CHANGES_REQUESTED** [review]
That means everywhere

**Comment by Tofandel** [comment]
Wow, that's some big breaking change from sass.. I mean they could have opted for another character to handle grids... Instead of breaking every single division out there.. It would have seemed like the more reasonable approach

Oh well

**Comment by mcdmaster** [comment]
> That means everywhere. If you want to propose testing changes that can be a separate pr

@KaelWD Done. Please check. Thanks

File: packages/vuetify/src/components/VGrid/_mixins.sass [hunk 3]
```diff
@@ -14,34 +14,34 @@
...
 
 =make-col-offset($size, $columns: $grid-columns)
-  $num: $size / $columns
+  $num: div($size, $columns)
   +ltr()
-    margin-left: if($num == 0, 0, percentage($num))
+    margin-left: if($num == 0, 0, calc(percentage($num)))
   +rtl()
-    margin-right: if($num == 0, 0, percentage($num))
+    margin-right: if($num == 0, 0, calc(percentage($num)))
```
Comment (KaelWD) [inline packages/vuetify/src/components/VGrid/_mixins.sass:38]: Why calc here?

File: packages/vuetify/src/components/VSwitch/VSwitch.sass [hunk 4]
```diff
@@ -112,7 +112,7 @@
       height: $switch-track-inset-height
       left: -4px
       opacity: $switch-track-inset-opacity
-      top: calc(50% - #{$switch-track-inset-height / 2})
+      top: 50% - div($switch-track-inset-height, 2)
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSwitch/VSwitch.sass:115]: Invalid

File: package.json [hunk 5]
```diff
@@ -67,6 +81,8 @@
     "sass": "^1.26.11",
```
Comment (KaelWD) [inline package.json:81]: Update so we get deprecation messages when we build

File: package.json [hunk 6]
```diff
@@ -25,6 +25,19 @@
     "node": "^12 || ^14",
     "yarn": "^1.12"
   },
+  "stylelint": {
+    "rules": {
+      "function-calc-no-invalid": [
```
Comment (KaelWD) [inline package.json:30]: This doesn't seem to do anything to sass judging by their docs.

File: package.json [hunk 7]
```diff
@@ -77,7 +93,6 @@
     "vue-router": "^3.0.1",
     "vue-template-compiler": "^2.6.11",
     "vuex": "^3.0.1",
-    "webpack": "^4.39.1",
```
[the remark this answers is not in the document]
Comment (mcdmaster) [reply package.json]: Seems like mistakenly removed

File: package.json (hunk 5 above)
Comment (mcdmaster) [reply package.json]: Current @^1.34.0

File: packages/vuetify/src/components/VProgressLinear/VProgressLinear.sass [hunk 8]
```diff
@@ -110,7 +110,7 @@
   opacity: $progress-linear-stream-opacity
   pointer-events: none
   position: absolute
-  top: calc(50% - #{$progress-linear-stream-border-width / 2})
+  top: 50% - div($progress-linear-stream-border-width, 2)
```
[the remark this answers is not in the document]
Comment (mcdmaster) [reply packages/vuetify/src/components/VProgressLinear/VProgressLinear.sass]: Please elaborate on

File: packages/vuetify/src/components/VSlider/VSlider.sass [hunk 9]
```diff
@@ -242,7 +242,7 @@
   .v-slider__thumb
     width: #{$slider-thumb-size - 4}
     height: #{$slider-thumb-size - 4}
-    left: -#{($slider-thumb-size - 4) / 2}
+    left: div(4 - $slider-thumb-size, 2)
```
[the remark this answers is not in the document]
Comment (mcdmaster) [reply packages/vuetify/src/components/VSlider/VSlider.sass:245]: Thank you!

File: packages/vuetify/src/components/VGrid/_mixins.sass (hunk 3 above)
Comment (mcdmaster) [reply packages/vuetify/src/components/VGrid/_mixins.sass]: Seems testing sludge was there...

File: packages/vuetify/src/components/VSwitch/VSwitch.sass (hunk 4 above)
Comment (mcdmaster) [reply packages/vuetify/src/components/VSwitch/VSwitch.sass]: Do elaborate on why again. Thanks

**Review by johnleider — CHANGES_REQUESTED** [review]
This PR contains multiple commits that have nothing to do with this bug fix. Can you please remove them.

File: packages/vuetify/src/components/VProgressLinear/VProgressLinear.sass (hunk 8 above)
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VProgressLinear/VProgressLinear.sass]: `calc` is runtime, `div` is compile-time. Everything that was originally using calc needs to continue doing so, you can't subtract px from % at compile-time.

**Comment by johnleider** [comment]
Looking good. What is the reason behind the CI changes?

**Comment by mcdmaster** [comment]
@johnleider Thank you. BTW, I was not confident with my CI credentials properly registered.
Can you please let me know if it is better to withdraw the last commit of mine?

**Review by KaelWD — CHANGES_REQUESTED** [review]
Build is failing and still shows some deprecation notices, please double-check your work.
